### ISSUE TITLE
About the batch size and optimizer of training

### ISSUE THREAD

**Issue by wsy-yjys** [huawei-noah/VanillaNet#13]
Question 1: Will the batch size during training affect the final performance of the model? Have you tried small batch size, such as 16 and 32? 

Question 2: Why do you use LAMB optimizer instead of SGD, Adamw and other more general optimizer? I use vanillanet as backbone for other downstream tasks with SGD optimizer but find it difficult to train the model from scratch，As you can see below, the first 50 epochs of model performance is basically no optimization, so I'd be grateful if you could give me some advice.

![image](https://github.com/huawei-noah/VanillaNet/assets/78601159/ad05fd38-4ae2-4784-8044-a75f92a5bcff)

**Comment by wsy-yjys** [comment]
Another question about the batch size, when I reduce the bacth size, does the learning rate need to be reduced correspondingly, thank you~

**Comment by ggjy** [comment]
Thank you for your interest in our work. For Q1 & Q3, we have only experimented with batch sizes of 1024 and 2048, and their results were similar. As the batch size varies, the learning rate needs to be adjusted accordingly, approximately by effective_lr=sqrt(effective_bs/base_bs) * base_lr. We have not attempted smaller batch sizes at this time.

As for Q2, on ImageNet, the results achieved by Adamw/LAMB is much better than SGD. The reason to finally use LAMB is similar to [this paper](https://arxiv.org/pdf/2110.00476.pdf), the LAMB is slighly better than Adamw. As for downstream tasks, I think Adamw can help improve the convergence speed of VanillaNet. You can also try to lower down the LR.

**Comment by PGthree3** [comment]
楼主你好，请问关于Q2从头训练下游任务，包括了深度训练策略吗

**Comment by PGthree3** [comment]
楼主你好，请问关于Q2从头训练下游任务，使用了深度训练策略吗

**Comment by wsy-yjys** [comment]
@PGthree3 是的

**Comment by wsy-yjys** [comment]
@ggjy Thank you for your reply~
